nikola/plugin_categories.py

```diff
@@ -372,6 +379,71 @@ class MarkdownExtension(CompilerExtension):
     compiler_name = "markdown"
 
 
+class MetadataExtractor(BasePlugin):
+    """Plugins that can extract meta information from post files."""
+
+    # Name of the extractor. (required)
+    name = "unknown"
+    # Where to get metadata from. (MetaSource; required)
+    source = None
+    # Priority of extractor. (MetaPriority; required)
+    priority = None
+    # List of tuples (MetaCondition, arg) with conditions used to select this extractor.
+    conditions = []
+    # Regular expression used for splitting metadata, or None if not applicable.
+    split_metadata_re = None
+    # List of tuples (import name, pip name, friendly name) of Python packages required for this extractor.
+    requirements = []
+    # Name of METADATA_MAPPING to use, if any.
+    map_from = None
+    # Whether or not the extractor supports writing metadata.
+    supports_write = False
+
+    def _extract_metadata_from_text(self, source_text: str) -> dict:
+        """Extract metadata from text."""
+        raise NotImplementedError()
+
+    def split_metadata_from_text(self, source_text: str) -> (str, str):
+        """Split text into metadata and content (both strings)."""
+        if self.split_metadata_re is None:
+            return source_text
+        else:
+            split_result = self.split_metadata_re.split(source_text.lstrip(), maxsplit=1)
+            if len(split_result) == 1:
+                return split_result[0], split_result[0]
+            else:
+                # Necessary?
+                return split_result[0], split_result[-1]
+
+    def extract_text(self, source_text: str) -> dict:
+        """Split file, return metadata and the content."""
+        split = self.split_metadata_from_text(source_text)
+        meta = self._extract_metadata_from_text(split[0])
+        return meta
+
+    def extract_filename(self, filename: str, lang: str) -> dict:
+        """Extract metadata from filename."""
+        return {}
+
+    def write_metadata(self, metadata: dict, comment_wrap=False) -> str:
```

Comment:
Better solution, change the behavior of Nikola’s extractor. A copy of a small dict shouldn’t hurt.